carts/views.py

```diff
@@ -1,3 +1,72 @@
-from django.shortcuts import render
+import json
+from json.decoder import JSONDecodeError
 
-# Create your views here.
+from django.http.response import JsonResponse
+from django.views         import View
+
+from users.utils     import LogInDecorator
+from carts.models    import Cart
+from products.models import ProductOption
+
+class CartView(View):
+    @LogInDecorator
+    def post(self, request):
+        try:
+            data       = json.loads(request.body)
+            user       = request.user
+            product_id = data['product_id']
+            size_id    = data['size_id']
+            quantity   = data['quantity']    # 값 추가할 경우 프론트에서 계산 후에 준 값으로 계산
+            
+            if not ProductOption.objects.filter(product_id=product_id, size_id=size_id).exists():
+                return JsonResponse({'ProductOption_DoesNotExist'}, status=404)
+            
+            product_option = ProductOption.objects.get(product_id=product_id, size_id=size_id)
+            
+            if quantity < 1 or quantity > int(product_option.stock):
+                return JsonResponse({'message': 'INVALID_QUANTITY'}, status=400)
+
+            cart, is_created     = Cart.objects.get_or_create(
+                user             = user,
+                products_options = product_option,
+                defaults         = {'quantity': 1}
+            )
+            cart.quantity = quantity
+            cart.save()
+            
+            return JsonResponse({'message': 'SUCCESS'}, status=201)   
+        except KeyError:
+            return JsonResponse({'message': 'KeyError'}, status=400)
+        except JSONDecodeError:
+            return JsonResponse({'message': 'JSONDecodeError'}, status=400)
+
+    @LogInDecorator
+    def get(self, request):
+        user   = request.user
+        carts  = Cart.objects.filter(user=user)
+        result = []
+        
+        for cart in carts:
+            product = cart.products_options.product
+            image   = product.productimage_set.filter().first()
+            
+            result.append(
+                {
+                    'cart_id'       : cart.id,
+                    'product_id'    : product.id,
+                    'product_name'  : product.name,
+                    'product_number': product.product_number,
+                    'price'         : int(product.price),
+                    'size'          : cart.products_options.size.name,
+                    'image_url'     : image.url,
+                    'quantity'      : cart.quantity
+                }
+            )
+        return JsonResponse({'resutl': result}, status=200)
+
+    @LogInDecorator
+    def delete(self, request, cart_id):
+        user = request.user
+        cart = Cart.objects.get(pk=cart_id)
```

Comment:
```suggestion
        user = request.user
        cart = Cart.objects.get(pk=cart_id, user=user)
```
1. 장바구니를 확인 할 때 전달 받은 장바구니의 pk와 user를 동시에 조건을 넣어 주어야 해당 장바구니가 로그인 한 user의 장바구니인지 확인이 됩니다.(질문 주신 것에 대한 답변도 되었을 것 같네요.)
2. get을 통해 데이터를 가져오고 있으니 get을 사용 했을 때 발생할 수 있는 에러처리도 같이 해주세요.

Reply:
경훈님 답변 감사합니다! 수정 후 바로 PR하겠습니다.